### PR TITLE
Put ssh keys on configured path in sshd_config (AuthorizedKeysFile)

### PR DESCRIPTION
**Pull request by mathieupotier**
I had an issue during deploying ssh keys they wasn't going into the correct path.

Signed-off-by: POTIER Mathieu mathieu.potier@onzeway.eu

### CODE REVIEW

**Comment by puneetk** [comment]
please add to defaults.yaml too

**Comment by mathieupotier** [comment]
I don't understand, I'm using the pillar data to get the path to authorized_keys file.
The defaults.yaml file is not managing this kind of settings in this formula.

I'm not enough confident with salt for the moment to review the whole formula.

**Comment by mathieupotier** [comment]
Have issue resolving the default value to %h/.ssh/authorized_keys... seems the %h is not taken in to account during evaluation, but working fine for /etc/ssh/AUTHKEYS/%u ... have to find a way to spec a default value which is working.

File: openssh/auth.sls [hunk 1]
```diff
@@ -38,12 +38,18 @@ include:
...
+    - config: {{ pillar['sshd_config']['AuthorizedKeysFile'] }}
+    {% endif -%}
     - require:
       - service: {{ openssh.service }}
     {%- else %}
 {{ print_name(identifier, key) }}:
   ssh_auth.absent:
     {{ print_ssh_auth(identifier, key) }}
+    {%- if 'sshd_config' in pillar and 'AuthorizedKeysFile' in pillar['sshd_config'] %}
+    - config: {{ pillar['sshd_config']['AuthorizedKeysFile'] }}
```
Comment (pepoluan) [inline openssh/auth.sls]: You need to put the `{{ ... }}` part between single quotes, like this: `'{{ ... }}'`

The reason is a bit complex:

- The `%` character has a special meaning in YAML
- On line 26 of `pillar.example`, the `%` sign is quoted. So it becomes a normal character.
- The whole line, without quotes, gets saved as the value of `pillar['sshd_config']['AuthorizedKeysFile']`
- However in the `auth.sls` file, the content of the above gets inlined _without quotes_, resulting in (after Jinja rendering):

        - config: %h/.ssh/authorized_keys

- And this last construct -- which has an unquoted `%` sign -- makes YAML go haywire.

**Comment by mathieupotier** [comment]
Can you please check again ?
Made some change to allow user to set a value with %h and %u

**Comment by mathieupotier** [comment]
PR opened since 1 year now.
I'm still wait for definitive feedback, can you please check ?

**Comment by gravyboat** [comment]
@mathieupotier I'm just going to merge this in. If someone has a problem they can create an issue.

**Pull request merged**